pkg/util/convert/convert.go

```diff
@@ -0,0 +1,87 @@
+package convert
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+
+	kudov1beta1 "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+// String returns a pointer to the string value passed in.
+func String(v string) *string {
+	return &v
+}
+
+// StringValue returns the value of the string pointer passed in or
+// "" if the pointer is nil.
+func StringValue(v *string) string {
+	if v != nil {
+		return *v
+	}
+	return ""
+}
+
+// ParamValue unwraps a parameter value.
+// Depending on the parameter type, the value can represent a string or an object described in YAML.
+func ParamValue(v *string, t kudov1beta1.ParameterType) (r interface{}, err error) {
+	switch t {
+	case kudov1beta1.MapValueType:
+		r, err = YAMLMap(StringValue(v))
+	case kudov1beta1.ArrayValueType:
+		r, err = YAMLArray(StringValue(v))
+	case kudov1beta1.StringValueType:
+		fallthrough
+	default:
+		r = StringValue(v)
+	}
+
+	return
+}
+
+// YAMLValue wraps a parameter value.
+func YAMLValue(i interface{}, t kudov1beta1.ParameterType) (*string, error) {
```

Comment:
This needs a different name. It's named similar to `YAMLArray` and `YAMLMap`, but the signature is the opposite. 

Maybe `convert.FromYAMLToString`?

Reply:
Good point, but still not right, because the interface isn't always YAML. Let me rename this as well as `ParamValue` accordingly.

Reply:
Renamed to `UnwrapParamValue` and `WrapParamValue`.